### ISSUE TITLE
'addApplicationDelegate' is unavailable in application extensions for iOS

### ISSUE THREAD

**Issue by offline-first** [rafaelsetragni/awesome_notifications#893]
# Rules

To report a bug and help us to provide the solution as fast as possible, please follow the instructions below:

## **1 - Describe the bug**

Failed to build iOS app
```console
Swift Compiler Error (Xcode): 'addApplicationDelegate' is unavailable in application extensions for iOS: Disallowed in plugins used in app extensions
/Users/arian/.pub-cache/hosted/pub.dev/awesome_notifications-0.8.2/ios/Classes/lib/SwiftAwesomeNotificationsPlugin.swift:57:22
```

## **2 - Provide us as much information as possible**

* Your plugins list inside the file `pubspec.yaml` (at least all notification plugins)
```
awesome_notifications_core: ^0.8.1
awesome_notifications: any  # <- this version will be managed by core plugin
awesome_notifications_fcm: any # <- this version will be managed by core plugin
```

* What platform did you found the issue
ios
* What devices did you found the issue
Simulator iPhone 14 Pro, iOS 16.4
* How are you initializing the plugin
As in the example
* How are you creating the notification 
As in the example
* The output of `flutter doctor` command
```[✓] Flutter (Channel stable, 3.16.0, on macOS 13.0.1 22A400 darwin-arm64, locale en-CY)
    • Flutter version 3.16.0 on channel stable at /Users/arian/dev/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision db7ef5bf9f (4 days ago), 2023-11-15 11:25:44 -0800
    • Engine revision 74d16627b9
    • Dart version 3.2.0
    • DevTools version 2.28.2

....
[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.1
...
```

## **3 - How To Reproduce the Error**

1. run debug

## **4 - Expected behavior**
...

## **5 - Screenshots**
![build_distribution](https://github.com/rafaelsetragni/awesome_notifications/assets/2144520/9065f65d-e7a9-4b12-9933-a00031d61129)

## **6 - Additional context**
For all TARGETS "Build Libraries for Distribution" is set to "No" in XCode.

**Comment by alanrubin** [comment]
Experiencing the same with the versions above.

**Comment by dairy96** [comment]
same here

**Comment by iabdousd** [comment]
@rafaelsetragni this is happening after upgrading to Flutter 3.16.

**Comment by tusharbhambere** [comment]
Any Solution?

**Comment by bashiransari** [comment]
Same issue here, any solutions ?

**Comment by ScottWallace** [comment]
Same here, dead stop on further builds. Need a solution.

**Comment by bashiransari** [comment]
Just use flutter 3.13.9 for now, until the issue is fixed.
I talked @rafaelsetragni that was the solution he gave me, at least unblocked me from development.

**Comment by ScottWallace** [comment]
Downgrading flutter is creating far too many package dependencies issues, since so many others (not this) have upgraded to work with 3.16. Having to rethink this package.

**Comment by abdalmonem** [comment]
@scopendo We also (as a company) started to thinking about stop using this plugin, We love it  and we hope them the best , and they do great job, but the support is really bad unfortunately.

**Comment by pks050505** [comment]
any solution---

**Comment by pks050505** [comment]
@offline-first have you solve this error?

**Comment by offline-first** [comment]
I've given up on it, setting up the FCM package for local development is too time consuming. I only know that the line `registrar.addApplicationDelegate(self)` in `SwiftAwesomeNotificationsPlugin.swift` is the problem. If you document this out, you can compile again but the app crashes as soon as you try to display a notification.

@rafaelsetragni  This bug is now three weeks old. This error is now three weeks old. Can you help us further?

**Comment by a1erons** [comment]
Continue using Flutter 3.13.9 till the issue is fixed. I trust a solution will be made for this. Just saw the same issue right after upgrading too

**Comment by ScottWallace** [comment]
Far too many packages are dependent on the latest flutter to go to that effort for a developer who won't make an effort for their own project, so, no, we've moved on.

Also, we considered paying the ten bucks at one point, but I can see now it would take weeeeks to actually get the license key after paying, so, no, again, we've moved on.

Walking backwards is not an option. Too bad, this was a great idea, and could have saved us a lot of hours. What a shame it's abandoned.

**Comment by scopendo** [comment]
@ScottWallace what did you use instead? Firebase Messaging with its limitations, developing your own plugins or something else?

**Comment by scalz** [comment]
@ScottWallace  snake biting its tails. There is an iOS compilation error, but using fcm package you'll get limitations on iOS without app extension.
It can be frustrating, but I dont think Rafael planned that flutter team would add features to iOS app extensions and break some of his work, so I think it's quite unfortunate, for us and also for Rafael.
So far I've not seen a better alternative solving ios limitations. Latest release is recent so it's definitely not abandoned. 
Sure, some communication about this issue would help, and reassure us.
I can't downgrade flutter either. In the meantime, I'm working on other platforms so I can go forward, and am waiting too for a solution.

**Comment by chadpav** [comment]
Guys, I think the iOS Service Extension is optional. There are a couple of nice features that it provides but they are nice-to-have features at best. For me, the only reason I needed it is to download file attachments in push notifications. On iOS, those are kind of hidden from most users anyway.

I see a few options here if I have to run Flutter Latest:
1) Remove the Awn service extension & either go without it for now or write my own that only does what I need.
2) The source is on Github, I assume @rafaelsetragni would take a PR to help him out. I could spend some time trying to understand his code and see how big the refactor would be. I think it's a matter of converting the extension to using Scene based API's instead of the App Delegate lifecycle API's.
3) Wait patiently and remember how little I paid for this product. Maybe think of it as a hybrid OSS project where the active maintainer needs a minimal amount to work on this. 

This is still the best Flutter notifications package available that solves a bunch of other cross-platform problems quite well.

@rafaelsetragni I think it would go a long way if you just acknowledged the bug and that it's on your list. If you are working on other stuff I think people would understand that you have resource limits.

**Comment by nietsmmar** [comment]
@chadpav I talked to @rafaelsetragni and he told me, that he is aware of the bug and is working on it. But I guess it can still take a while.

**Comment by scopendo** [comment]
> @rafaelsetragni I think it would go a long way if you just acknowledged the bug and that it's on your list. If you are working on other stuff I think people would understand that you have resource limits.

This is the key point – I have every respect for OSS maintainers but at this time I can't tell whether this is a project that is still being maintained, bearing this isn't a minor issue.

Perhaps @rafaelsetragni might consider submitting this to https://github.com/fluttercommunity to see if the burden can be shared by the wider community.

**Comment by rafaelsetragni** [comment]
Hello everyone,

Thank you for your patience and understanding, regarding the recent Flutter updates. I'm aware of the challenges caused by the totally unexpected deprecation of addApplicationDelegate for target extensions. All other deprecations was correctly address, except this one. Due to limited resources, addressing this issue took longer than what i anticipated.

But i have good news: I'm finalizing an update for Flutter 3.16, expected to release in a few days. Until then, using Flutter 3.13.9 is the best temporary solution.

I appreciate the community's support and suggestions.
Also, contributions through pull requests are always welcome.

**Comment by scopendo** [comment]
Thank you for the update @rafaelsetragni and speaking for myself, there is no need to apologise for it taking so long, it's just the not knowing of whether it is even going to be worked on.

Is there a detailed technical guide that describes the plugin implementation? I think the problem I and perhaps others have is lacking the iOS skills to maintain such a plugin, in order to contribute.

**Comment by rafaelsetragni** [comment]
Yes, we are on version 0.9.2

**Comment by ScottWallace** [comment]
> Yes, we are on version 0.9.2. This issue was already fixed.

Where? There's no release 0.9.2, the release branch isn't 0.9.2, the hotfix branch isn't 0.9.2, nor development branch. Seriously, where is this fixed in a way that allows people to 1) access it, and 2) trust it.

**Comment by scopendo** [comment]
https://pub.dev/packages/awesome_notifications/changelog
https://pub.dev/packages/awesome_notifications_fcm/changelog

**Comment by ScottWallace** [comment]
> https://pub.dev/packages/awesome_notifications/changelog https://pub.dev/packages/awesome_notifications_fcm/changelog


Again, where is this code? Not a single branch or release on GitHub (where we are now) reflects this. Where is this really?

**Comment by nietsmmar** [comment]
> > https://pub.dev/packages/awesome_notifications/changelog https://pub.dev/packages/awesome_notifications_fcm/changelog
> 
> Again, where is this code? Not a single branch or release on GitHub (where we are now) reflects this. Where is this really?

@ScottWallace I understand your frustration. Probably you need to get something fixed and are in a hurry and really want to have better communication. I feel a bit sad reading your comments though as they seem a bit aggressive. This is still mainly open source code so I would really like to keep the discussions respectful. We shouldn't be demanding too much.

The flutter fix is already in this branch https://github.com/rafaelsetragni/awesome_notifications/tree/hotfixes/0.9.0-fixes

Although this is not the newest version.

**Comment by ScottWallace** [comment]
...complains about my "frustration" then makes my point. "Although this is not the newest version," is true, and you'll plainly see in the URL that you point to something other than what you say is available. This is not the 0.9.2 "release" which seems not to exist in code. You call it that on pub.dev, but it's clearly not - that's an untrue 'release' from you on pub.dev.

There is no 0.9.2 release in source code. Please tell the truth.

Also, you're **not open source**. You're charging **$10 for full functionality**. Don't hide behind calling it open source when it's a **paid product**, and don't be surprised when **paying customers** don't appreciate you disappearing from a **clearly broken** paid product, then not be truthful about the "fix," which isn't available in code.

No, we're not relying on your **paid product** (not open source) because, having read YOUR comments, you've been far less than truthful, about releases, about capabilities, about fixes, and about whether or not you're open source.

This is disgraceful.

**Comment by nietsmmar** [comment]
> ...complains about my "frustration" then makes my point. "Although this is not the newest version," is true, and you'll plainly see in the URL that you point to something other than what you say is available. This is not the 0.9.2 "release" which seems not to exist in code. You call it that on pub.dev, but it's clearly not - that's an untrue 'release' from you on pub.dev.
> 
> There is no 0.9.2 release in source code. Please tell the truth.
> 
> Also, you're **not open source**. You're charging **$10 for full functionality**. Don't hide behind calling it open source when it's a **paid product**, and don't be surprised when **paying customers** don't appreciate you disappearing from a **clearly broken** paid product, then not be truthful about the "fix," which isn't available in code.
> 
> No, we're not relying on your **paid product** (not open source) because, having read YOUR comments, you've been far less than truthful, about releases, about capabilities, about fixes, and about whether or not you're open source.
> 
> This is disgraceful.

I understand that you are feeling quite bad at the moment. I am sorry. I did not want to complain about your frustrating. Actually I wanted to tell you, that I even understand your frustration.

Nor did I say that the 0.9.2 version is published or anywhere. I said, that the fix is in this branch. The fix that this issue discussion is about. I even said myself, that it is not the newest version. I just wanted to help you out and give you the only link I know of where the fix for the issue discussed here is published.

I am not having anything to do with this project. I am just a random person from the internet using this code. It is still mostly open-source as it works without having a license. And you said, you want to have it working? Why not using the pub.dev version then? It works for me. This bug is fixed there. So I don't see why this is clearly broken. 

You say I am not truthfull of MY product? This is not even my product. I really don't know why you are so angry. You are clearly under a lot of stress. I am sorry that there is so much pressure on you to build your project. I sincerely hope there will be easier times soon.
